balena/models/device.py

```diff
@@ -1087,9 +1089,14 @@ def generate_device_key(self, uuid):
         """
 
         device_id = self.get(uuid)['id']
+        
+        data = {
+            'name': key_name,
+            'description': key_description
```

Comment:
You can also add the expiryDate if you want, but I think it should be defaulting to Null rather than empty string.
```suggestion
            'description': key_description,
            'expiryDate': expiry_date
```